app/models/ability.rb

```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.present?
+      can %i(create read update destroy), Journal
+      can %i(create read update destroy), JournalEntry
+      can %i(create read), User
+      can :update, User, id: user.id
+
+      if user.admin?
+        can :manage, :all
+      end
+    end
+  end
```

Comment:
```suggestion
  def initialize(user)
    return if user.blank?
 
    if user.admin?
       can :manage, :all
    else
      can %i(create read update destroy), [Journal, JournalEntry]
      can %i(create read), User
      can :update, User, id: user_id
    end
  end
```

Reply:
You may want to learn about the cannot feature.
Read more from here: https://github.com/CanCanCommunity/cancancan/blob/develop/docs/README.md